File: tests/benchmark/Call_Move.cpp
#include "timer.h"
#include <iostream>

class NPC
{
public:
  NPC() {x=y=z = vx=vy=vz = 0.0;}
  void move(double new_vx, double new_vy, double new_vz,
            double &res_x, double &res_y, double &res_z,
            double &res_vx, double &res_vy, double &res_vz);
private:
  double x,y,z;
  double vx,vy,vz;
};

void NPC::move(double new_vx, double new_vy, double new_vz,
               double &res_x, double &res_y, double &res_z,
               double &res_vx, double &res_vy, double &res_vz)
{
  vx = new_vx;
  vy = new_vy;
  vz = new_vz;
  
  x += vx;
  y += vy;
  z += vz;

  res_x = x;
  res_y = y;
  res_z = z;

  res_vx = vx;
  res_vy = vy;
  res_vz = vz;
}

int main()
{
  double i;
  NPC npc1;
  double x,y,z, vx,vy,vz;
  TIME_ON;
  for(i=0; i<10000000.0; i+=1.0)
    npc1.move(i,i-1.0,i+1.0, x,y,z, vx,vy,vz);
  TIME_OFF("NPC movements");
  std::cout<<"Resulting position: ("<<x<<","<<y<<","<<z<<")"<<std::endl;
  return 0;
}

File: benchmark/Call_Move.py
from timer import *

class NPC:
    def __init__(self):
        self.x = 0.0
        self.y = 0.0
        self.z = 0.0
        self.vx = 0.0
        self.vy = 0.0
        self.vz = 0.0
    def move(self, vx, vy, vz):
        self.vx, self.vy, self.vz = vx, vy, vz
        self.x = self.x + self.vx
        self.y = self.y + self.vy
        self.z = self.z + self.vz
        return self.x, self.y, self.z

maxi=100000
npc1 = NPC()
time_on()
for i in range(maxi):
    x,y,z = npc1.move(i+0.0, i-1.0, i+1.0)
time_off("NPC movements", maxi)
print "Resulting position: (%s,%s,%s)" % (x,y,z)
